docs/source/tutorials/token_classification/utils.py

```diff
@@ -0,0 +1,242 @@
+                j_new = 0 
+        
+        if i != i_new: 
+            probs = np.average(scores[i:i_new], axis=0, weights=weights if weighted else None) 
+        else: 
+            probs = scores[i] 
+        i, j = i_new, j_new 
+        
+        pred_probs.append(probs) 
+        labels.append(label)
+        
+    return np.array(pred_probs), labels 
+
+def to_dict(nl): 
+    return {str(i): l for i, l in enumerate(nl)} 
+
+def read_npz(filepath): 
+    data = dict(np.load(filepath)) 
+    data = [data[str(i)] for i in range(len(data))] 
+    return data 
+
+def color_sentence(sentence, word): 
+    start_idx = sentence.index(word) 
+    before, after = sentence[:start_idx], sentence[start_idx + len(word):]
+    return '%s%s%s' % (before, colored(word, 'red'), after) 
+
+
+def show_issues(issues, labels, pred_probs, given_words, sentences, 
+                top=10, entities=None, exclude=[]): 
+    if not entities: 
+        entities = ['O', 'MISC', 'PER', 'ORG', 'LOC'] 
+    def show_issue(issue): 
+        i, j = issue 
+        issue_label = labels[i][j] 
+        predicted_label = np.argmax(pred_probs[i][j]) 
+        issue_word = given_words[i][j] 
+        issue_sentence = color_sentence(sentences[i], issue_word) 
+        return issue_sentence, (issue_label, predicted_label)  
+    
+    shown = 0 
+    for issue in issues: 
+        sentence, (given, predicted) = show_issue(issue) 
+        if (given, predicted) in exclude: 
+            continue 
+        print('%d. %s' % (shown+1, sentence)) 
+        print('Given label: %s, predicted label: %s\n' % (entities[given], entities[predicted])) 
+        shown += 1 
+        if shown == top: 
+            break 
+
+def common_token_issues(issues, given_words, labels, pred_probs, entities, 
+                        top=10, exclude=[], verbose=True): 
+    n = pred_probs[0].shape[1] 
+    count = {} 
+    for issue in issues: 
+        i, j = issue 
+        word = given_words[i][j] 
+        label = labels[i][j] 
+        pred = pred_probs[i][j].argmax() 
+        if word not in count: 
+            count[word] = np.zeros([n, n], dtype=int) 
+        if (label, pred) not in exclude: 
+            count[word][label][pred] += 1 
+    words = [word for word in count.keys()] 
+    n = count[words[0]].shape[0] 
+    freq = [np.sum(count[word]) for word in words] 
+    rank = np.argsort(freq)[::-1][:top] 
+    
+    for r in rank: 
+        word = words[r] 
+        matrix = count[word] 
+        most_frequent = np.argsort(matrix.flatten())[::-1] 
+        print("'%s' is mislabeled %d times" % (word, freq[r])) 
+        if verbose: 
+            print('-----------------------------') 
+            for f in most_frequent: 
+                i, j = f // n, f % n 
+                if matrix[i][j] == 0: 
+                    break 
+                print('labeled as %s, but predicted as %s %d times' % 
+                      (entities[i], entities[j], matrix[i][j])) 
+        print() 
```

Comment:
Let's have this function return dataframe that includes the info printed here, the dataframe should look like this:

token    given_label    true_label   num_label_issues
  /               O                    LOC               36
Chicago    ORG               LOC               27
  /               O                    PER                4
  ....
  
  where this dataframe is sorted by the number of times each type of label-flip has occurred.

Reply:
I'm assuming we're returning the full list right? (instead of the top `top` mislabels)

Reply:
Probably better to name it `predicted_label` instead of `true_label`. We don't really know the ground truth

Reply:
Calling it `predicted_label` is also not ideal IMO because then this table looks very similar to a standard confusion matrix and it's hard to know the difference (this table is only summarizing a small subset of the occurrences of each token, where we think there was a label issue).  How about `inferred_true_label` or `suggested_label` instead?